pkg/terraform/exec/plugins/vsphereprivate/config.go

```diff
@@ -0,0 +1,571 @@
+// Modified From:
+// https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/config.go
```

Comment:
this file is already available as public in the `terraform-provider-vsphere` so why copy?

Reply:
@abhinavdahiya we need `vimClient` from the `VSphereClient` struct.
https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/config.go#L35

If there is a better way please let me know because I certainly didn't want to add more here than we need.

Reply:
```go
type VSphereClient struct {
	// The VIM/govmomi client.
	vimClient *govmomi.Client
}

type ConfigWrapper struct {
	config *vsphere.Config
}

func NewConfig(d *schema.ResourceData) (*ConfigWrapper, error) {
	config, err := vsphere.NewConfig(d)
	if err != nil {
		return nil, err
	}
	return &ConfigWrapper{config}
}

// vimURL returns a URL to pass to the VIM SOAP client.
func (cw *ConfigWrapper) vimURL() (*url.URL, error) {
	u, err := url.Parse("https://" + cw.config.VSphereServer + "/sdk")
	if err != nil {
		return nil, fmt.Errorf("error parse url: %s", err)
	}

	u.User = url.UserPassword(c.User, c.Password)

	return u, nil
}


// Client returns a new client for accessing VMWare vSphere.
func (cw *ConfigWrapper) Client() (*vsphere.VSphereClient, error) {
	client := new(VSphereClient)

	u, err := c.vimURL()
	if err != nil {
		return nil, fmt.Errorf("error generating SOAP endpoint url: %s", err)
	}

	err = c.config.EnableDebug()
	if err != nil {
		return nil, fmt.Errorf("error setting up client debug: %s", err)
	}

	// Set up the VIM/govmomi client connection, or load a previous session
	client.vimClient, err = c.config.SavedVimSessionOrNew(u)
	if err != nil {
		return nil, err
	}

	log.Printf("[DEBUG] VMWare vSphere Client configured for URL: %s", c.VSphereServer)

	return client, nil
}
```